tests/integration/elder.test.ts

```diff
@@ -0,0 +1,79 @@
+import Elder, {
+  Model,
+  StringType,
+  type,
+  Adapter,
+  PostgresAdapter,
+  Collection
+} from '../../src'
+
+const config = { adapters: { default: { database: 'ash' } } }
+
+describe('Elder', () => {
+  test('model setup', async () => {
+    class CatModel extends Model {
+      @type('string') name: string
+    }
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel }
+    })
+    const cats = await CatModel.all()
+    expect(cats).toBeInstanceOf(Collection)
+    expect(cats[0]).toBeInstanceOf(CatModel)
+    expect(cats[0].name).toBe('fluffy')
+    return orm.destroy()
+  })
+
+  test('model with custom type', async () => {
+    class CatModel extends Model {
+      @type('cat-name') name: string
+    }
+    class CatName extends StringType {}
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel },
+      types: { 'cat-name': CatName }
+    })
+    const cats = await CatModel.all()
+    expect(cats[0].name).toBe('fluffy')
+    return orm.destroy()
+  })
+
+  test('model with custom scoped type', async () => {
+    class CatModel extends Model {
+      @type('name') name: string
+    }
+    class CatName extends StringType {}
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel },
+      types: { 'cat:name': CatName }
```

Comment:
I don't know what this does, and the assertion doesn't make that pop out for me (similar to the above); i presume this is an alt syntax for the same thing you had above?

Reply:
ah, if that's the case then maybe add a DogModel or CatmanduModel and ensure they can/can't see the name field in the test?